tests/artifact/test_local_artifact.py

```diff
@@ -0,0 +1,254 @@
+    assert result is not None
+    assert sorted(result) == sorted(expectation)
+
+
+@pytest.mark.parametrize(
+    ("purl_str"),
+    [
+        pytest.param(
+            "pkg:pypi/django",
+            id="A pypi PURL without version",
+        ),
+        pytest.param(
+            "pkg:maven/com.google.guava/guava@33.2.1-jre",
+            id="The purl type is not supported.",
+        ),
+    ],
+)
+def test_construct_local_artifact_paths_glob_pattern_pypi_purl_error(purl_str: str) -> None:
+    """Test constructing a local artifact patterns from a given pypi purl with error."""
+    pypi_purl = PackageURL.from_string(purl_str)
+    result = construct_local_artifact_paths_glob_pattern_pypi_purl(pypi_purl=pypi_purl)
+    assert result is None
+
+
+def test_find_artifact_paths_from_invalid_python_venv() -> None:
+    """Test find_artifact_paths_from_python_venv method with invalid venv path"""
+    assert not find_artifact_paths_from_python_venv("./does-not-exist", ["django", "django-5.0.6.dist-info"])
+
+
+@pytest.mark.parametrize(
+    ("purl_str", "build_tool_purl_types", "local_artifact_repo_mapper", "expectation"),
+    [
+        pytest.param(
+            "pkg:maven/com.google.guava/guava@33.2.1-jre",
+            ["maven", "pypi"],
+            {},
+            {},
+            id="A maven type PURL where multiple build tool types are discovered. But no local repository is available.",
+        ),
+        pytest.param(
+            "pkg:maven/com.google.guava/guava@33.2.1-jre",
+            [],
+            {},
+            {},
+            id="A maven type PURL where no build tool types are discovered and no local repository is available.",
+        ),
+    ],
+)
+def test_get_local_artifact_paths_empty(
+    purl_str: str,
+    build_tool_purl_types: list[str],
+    local_artifact_repo_mapper: dict[str, str],
+    expectation: dict[str, list[str]],
+) -> None:
+    """Test getting local artifact paths where the result is empty."""
+    purl = PackageURL.from_string(purl_str)
+    assert (
+        get_local_artifact_paths(
+            purl=purl,
+            build_tool_purl_types=build_tool_purl_types,
+            local_artifact_repo_mapper=local_artifact_repo_mapper,
+        )
+        == expectation
+    )
+
+
+@pytest.mark.parametrize(
+    ("purl_str", "build_tool_purl_types", "expectation"),
+    [
+        pytest.param(
+            "pkg:maven/com.google.guava/guava@33.2.1-jre",
+            ["maven", "pypi"],
+            {},
+            id="A maven type PURL where multiple build tool types are discovered",
+        ),
+        pytest.param(
+            "pkg:maven/com.google.guava/guava@33.2.1-jre",
+            [],
+            {},
+            id="A maven type PURL where no build tool is discovered",
+        ),
+        pytest.param(
+            "pkg:pypi/django@5.0.3",
+            [],
+            {},
+            id="A maven type PURL where no build tool is discovered",
```

Comment:
```suggestion
            id="A pypi type PURL where no build tool is discovered",
```

Reply:
Fixed in https://github.com/oracle/macaron/pull/864/commits/da0f514376752d38af26e2be396c6e31001a7a24